### ISSUE TITLE
Detected clobbering during Lint/EmptyConditionalBody

### ISSUE THREAD

**Issue by SaschaSchwarz** [rubocop/rubocop#11003]
RuboCop fails to parse a class with an empty conditional body.  
This seems to be a variation of #10951.

--------

## Expected behavior

I am expecting a warning for the empty conditional body.

## Actual behavior
```
An error occurred while Lint/EmptyConditionalBody cop was inspecting /workspaces/saschaschwarz/test.rb:2:2.
```
<details><summary>complete log</summary>
<p>

```
$ rubocop --only Lint/EmptyConditionalBody -d test.rb 
For /workspaces/saschaschwarz: Default configuration from /usr/local/rvm/gems/default/gems/rubocop-1.36.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /workspaces/saschaschwarz/test.rb
An error occurred while Lint/EmptyConditionalBody cop was inspecting /workspaces/saschaschwarz/test.rb:2:2.
Parser::Source::TreeRewriter detected clobbering
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter.rb:427:in `trigger_policy'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter.rb:414:in `enforce_policy'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:233:in `call'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:233:in `swallow'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:97:in `with'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:124:in `place_in_hierarchy'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:106:in `do_combine'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:117:in `place_in_hierarchy'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:106:in `do_combine'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter/action.rb:30:in `combine'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter.rb:400:in `combine'
/usr/local/rvm/gems/default/gems/parser-3.1.2.1/lib/parser/source/tree_rewriter.rb:194:in `replace'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/lint/empty_conditional_body.rb:106:in `correct_other_branches'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/lint/empty_conditional_body.rb:84:in `autocorrect'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/lint/empty_conditional_body.rb:75:in `block in on_if'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/base.rb:346:in `correct'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/lint/empty_conditional_body.rb:74:in `on_if'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:69:in `on_if'
/usr/local/rvm/gems/default/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:153:in `on_class'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/usr/local/rvm/gems/default/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:83:in `investigate'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:315:in `inspect_file'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `loop'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:137:in `file_offenses'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:128:in `process_file'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `reduce'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:94:in `inspect_files'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/runner.rb:47:in `run'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/command.rb:11:in `run'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli/environment.rb:18:in `run'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli.rb:72:in `run_command'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli.rb:79:in `execute_runners'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/lib/rubocop/cli.rb:48:in `run'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/exe/rubocop:19:in `block in <top (required)>'
/usr/local/rvm/gems/default/gems/benchmark-0.2.0/lib/benchmark.rb:311:in `realtime'
/usr/local/rvm/gems/default/gems/rubocop-1.36.0/exe/rubocop:19:in `<top (required)>'
/usr/local/rvm/gems/default/bin/rubocop:23:in `load'
/usr/local/rvm/gems/default/bin/rubocop:23:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/EmptyConditionalBody cop was inspecting /workspaces/saschaschwarz/test.rb:2:2.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.6) [x86_64-linux]
Finished in 0.12385733300016 seconds
```

</p>
</details>

## Steps to reproduce the problem
` rubocop --only Lint/EmptyConditionalBody -d test.rb` with `test.rb` being

``` ruby
class Test
  if condition
  else
    def some_method
    end
  end
end
```

## RuboCop version
```
$ rubocop -V
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 2.7.6) [x86_64-linux]
```

**Comment by Earlopain** [comment]
Other/Smaller test case.

```rb
class Test
  if condition
  else
    do_something
  end
end
```

**Comment by srcoley** [comment]
I have submitted a PR to resolve this issue(#11087).
